Source/WebKit/Shared/DocumentEditingContext.serialization.in

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2023 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#if PLATFORM(IOS_FAMILY)
+
+[Nested] enum class WebKit::DocumentEditingContextRequest::Options : uint8_t {
+    Text,
+    AttributedText,
+    Rects,
+    Spatial,
+    Annotation,
+    MarkedTextRects,
+    SpatialAndCurrentSelection,
+    AutocorrectedRanges,
+};
+
+[Nested,OptionSet] enum class WebKit::DocumentEditingContextRequest::Options : uint8_t {
+    Text,
+    AttributedText,
+    Rects,
+    Spatial,
+    Annotation,
+    MarkedTextRects,
+    SpatialAndCurrentSelection,
+    AutocorrectedRanges,
+};
+
+[Nested] struct WebKit::DocumentEditingContext::Range {
+    uint64_t location;
+    uint64_t length;
```

Comment:
We should probably add a validator that location + length doesn't overflow:
e.g. [Validator='!(Checked<uint64_t> { *location } + *length).hasOverflowed()'] length.

Reply:
Rebased commit with main and added the suggested validator.